### PR TITLE
[Tokenizers] Add Space Clean Up

### PR DESCRIPTION
**Pull request by apaniukov**
Add `clean_up_tokenization_spaces` option to `convert_tokenizer` that mimics `huggingface_tokenizer.decode(token_ids, clean_up_tokenization_spaces=True | False)` behaviour.
Format code with ruff.
Remove unused includes.
Patch NodeFactory `__init__`.
Avoid _factory creation if not needed.
Add Fuzz Testing.

### CODE REVIEW

**Review by mryzhov — APPROVED** [review]

File: modules/custom_operations/user_ie_extensions/tokenizer/bytes_to_chars.cpp [hunk 1]
```diff
@@ -7,10 +7,6 @@
 #    pragma warning(disable : 4275)
 #endif
```
Comment (ilya-lavrenov) [inline modules/custom_operations/user_ie_extensions/tokenizer/bytes_to_chars.cpp:8]: I suppose we need to remove `warning(disable`  then as a WA for Fast Tokenizers headers

File: modules/custom_operations/user_ie_extensions/tokenizer/chars_to_bytes.cpp [hunk 2]
```diff
@@ -7,10 +7,6 @@
 #    pragma warning(disable : 4275)
 #endif
```
Comment (ilya-lavrenov) [inline modules/custom_operations/user_ie_extensions/tokenizer/chars_to_bytes.cpp:8]: the same

File: modules/custom_operations/user_ie_extensions/tokenizer/python/README.md [hunk 3]
```diff
@@ -278,302 +280,309 @@ This report is autogenerated and includes tokenizers and detokenizers tests. The
...
-      <td >39</td>
+      <td >95.00</td>
+      <td >40</td>
     </tr>
     <tr>
       <td >WordPiece</td>
       <td >jhgan/ko-sbert-sts</td>
-      <td >79.49</td>
-      <td >39</td>
+      <td >75.00</td>
```
Comment (ilya-lavrenov) [inline modules/custom_operations/user_ie_extensions/tokenizer/python/README.md:547]: `WordPiece` became worse, why?

Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/python/README.md:547]: I added a new test case with a control char that the current regex normalizer does not cover.

File: modules/custom_operations/user_ie_extensions/tokenizer/bytes_to_chars.cpp (hunk 1 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/bytes_to_chars.cpp]: Removed

File: modules/custom_operations/user_ie_extensions/tokenizer/chars_to_bytes.cpp (hunk 2 above)
Comment (apaniukov) [reply modules/custom_operations/user_ie_extensions/tokenizer/chars_to_bytes.cpp]: Removed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**